.github/workflows/build.yml

```diff
@@ -0,0 +1,44 @@
+name: CD
+
+on:
+  push:
+    branches:
+      - '*'
+    tags:
+      - 'v*.*.*' # Enforce Semantic Versioning
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+
+    - name: Setup CI Environment
+      uses: yetanalytics/actions/setup-env@v0
+
+    - name: Build Bundle
+      run: make bundle
+
+    - name: Archive Bundle (Branch Pushes)
+      if: ${{ startsWith(github.ref, 'refs/heads') }}
+      uses: actions/upload-artifact@v2
+      with:
+        name: lrs-admin-ui-artifact-${{ github.sha }}
+        path: target/bundle/**
+
+    - name: Compress Bundle
+      if: ${{ startsWith(github.ref, 'refs/tags') }}
+      run: | # Need to cd so that the zip file doesn't contain the parent dirs
+        cd target/bundle
+        zip -r ../../lrs-admin-ui.zip ./
+
+    - name: Craft Draft Release (Tag Pushes)
+      if: ${{ startsWith(github.ref, 'refs/tags') }}
+      uses: softprops/action-gh-release@v1
+      with:
+        # Defaults:
+        # name: [tag name]
+        # tag_name: github.ref
+        body: "## Release Notes\nTODO: Create great release notes!"
+        draft: true
+        files: lrs-admin-ui.zip
```

Comment:
With our new multi-action repo, we can create separate actions for building on branch pushes and building on tag pushes.

Reply:
Yep, I think more research is required. You can certainly provide parameterized composite actions for building, which would limit the boiler plate to everything outside the steps themselves at least.

For everything else like `on`, matrices, etc. We can use the templating approach via the `.github` repo, so then customization for new projects is just a click followed by minimal edits.